.github/workflows/ci.yml

```diff
@@ -38,3 +38,16 @@ jobs:
     - name: Django Testing project
       run: |
         pytest
+    - name: Generate badges
+      if: github.ref == 'refs/heads/master'
+      run : |
+        genbadge coverage -i docs/reports/coverage/coverage.xml -o docs/reports/coverage/coverage-badge.svg
+        genbadge tests -i  docs/reports/tests/pytest.xml -o docs/reports/tests/pytest-badge.svg
+    - name: Commit reports and badge
+      if: github.ref == 'refs/heads/master'
+      uses: stefanzweifel/git-auto-commit-action@v5
+      with:
+        commit_message: Pytest and Coverage Reports
```

Comment:
this creates a master commit when it runs ?! wouldn't that mean that there is lots of commit spam on master?

Reply:
The coverage reports would only run during CI on master, so would limit commits to just releases.

Reply:
thanks for the clarification .. to be completely honest I am really not sure what to think about this, I don't super like it this way ... 

How useful are the coverage results, is it actually possible to find spots where test coverage is bad?
also who will write the tests we know are missing, if I look at the github issues and also the things I just want to do I dont really think I will ever have the time to actually make anything out of these results while I will likely have to fix something every 1-2 years when it breaks for some unknown reason.

Reply:
The coverage reports are fully annotated so are helpful.  There is an example [here](https://smilerz.github.io/recipes/reports/coverage/)

That being said, there are other options. Setup coverage so it's only ever produced locally.  There is an online service that will host coverage reports and provide a badge (shields.io).

Reply:
thank you for the input, I will have to think about this but this code is pretty detatched so should not break for a while when working on other stuff.